### PR TITLE
JAMES-3400 James Webadmin Cli Domain Exist CommandFeature domainexist

### PR DESCRIPTION
**Pull request by chibenwa**
Reopens https://github.com/linagora/james-project/pull/3997

### CODE REVIEW

**Comment by chibenwa** [comment]
Always use JAMES-3400 as a JIRA commit message prefix

File: server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java [hunk 1]
```diff
@@ -121,4 +121,26 @@ void domainDeleteCommandWithDefaultDomainShouldFailed(GuiceJamesServer server) {
         assertThat(exitCode).isEqualTo(1);
     }
 
+    @Test
+    void domainExistCommandWithDefaultDomainShouldBeExisted(GuiceJamesServer server) {
```
Comment (Arsnael) [inline server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java:125]: `domainExistCommandWithDefaultDomainShouldSucceed`
or 
`domainExistCommandWithDefaultDomainShouldExist`

File: server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java [hunk 2]
```diff
@@ -121,4 +121,26 @@ void domainDeleteCommandWithDefaultDomainShouldFailed(GuiceJamesServer server) {
...
+
+        int exitCode = WebAdminCli.executeFluent(new PrintStream(outputStreamCaptor), new PrintStream(errorStreamCaptor),
+            "--url", "http://127.0.0.1:" + port.getValue(), "domain", "exist", "localhost");
+
+        assertThat(exitCode).isEqualTo(0);
+        assertThat(outputStreamCaptor.toString().trim()).isEqualTo("localhost exists");
+    }
+
+    @Test
+    void domainExistCommandNotWithDefaultDomainShouldNotBeExisted(GuiceJamesServer server) {
```
Comment (Arsnael) [inline server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java:136]: `domainExistCommandWithNonExistingDomainShouldFail`

File: server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java [hunk 3]
```diff
@@ -121,4 +121,26 @@ void domainDeleteCommandWithDefaultDomainShouldFailed(GuiceJamesServer server) {
...
+        Port port = server.getProbe(WebAdminGuiceProbe.class).getWebAdminPort();
+
+        int exitCode = WebAdminCli.executeFluent(new PrintStream(outputStreamCaptor), new PrintStream(errorStreamCaptor),
+            "--url", "http://127.0.0.1:" + port.getValue(), "domain", "exist", "linagora.com");
+
+        assertThat(exitCode).isEqualTo(0);
+        assertThat(outputStreamCaptor.toString().trim()).isEqualTo("linagora.com does not exist");
+    }
```
Comment (Arsnael) [inline server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java:144]: How about adding a test when you add a new domain, then check it exists?

**Comment by quantranhong1999** [comment]
https://github.com/quantranhong1999/james-project/tree/feature-domainexist-quanth

@chibenwa

File: server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java (hunk 1 above)
Comment (quantranhong1999) [reply server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java:125]: Solved

File: server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java (hunk 2 above)
Comment (quantranhong1999) [reply server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java:136]: Solved

File: server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java (hunk 3 above)
Comment (quantranhong1999) [reply server/protocols/webadmin-cli/src/test/java/org/apache/james/cli/DomainManageTest.java:144]: Solved

**Comment by LanKhuat** [comment]
[8103df2bb8907af32c9b228ae7031c09266defcd] [ERROR] Failures: 
[8103df2bb8907af32c9b228ae7031c09266defcd] [ERROR]   CassandraMailboxManagerTest$DeletionTests.deleteMailboxShouldCleanupACLWhenRightsDeleteFails:595 Multiple Failures (1 failure)
[8103df2bb8907af32c9b228ae7031c09266defcd] 	
[8103df2bb8907af32c9b228ae7031c09266defcd] Expecting empty but was:<[(CassandraId{id=a0d1abc0-1ffb-11eb-8df4-e9aed182527d},r)]>
[8103df2bb8907af32c9b228ae7031c09266defcd] at CassandraMailboxManagerTest$DeletionTests.lambda$deleteMailboxShouldCleanupACLWhenRightsDeleteFails$13(CassandraMailboxManagerTest$DeletionTests.java:600)

**Comment by chibenwa** [comment]
test this please

**Review by Arsnael — APPROVED** [review]

**Review by chibenwa — COMMENTED** [review]
:+1:

**Pull request merged**